splipy/basis_eval.pyx

```diff
@@ -0,0 +1,122 @@
+from bisect import bisect_right, bisect_left
+import numpy as np
+cimport numpy as np
+import copy
+cimport cython
+from cpython cimport bool
+@cython.boundscheck(False) # turn off bounds-checking for entire function
```

Comment:
This only disables bounds checks in the int_max function. You need to move it down to evaluate. Put one on snap too, maybe.

Reply:
Oh!? That is super important. Good catch!